src/test/java/bio/terra/service/filedata/DrsServiceTest.java

```diff
@@ -226,11 +222,195 @@ public void testLookupNegative() {
         () -> drsService.lookupObjectByDrsId(authUser, azureDrsObjectId, false));
   }
 
+  @Test
+  public void testLookupWithSeveralBillingSnapshots() throws InterruptedException {
```

Comment:
Another piece of food for thought: this unit test does a lot of heavy lifting.  In my own testing I've found this to be a frequent indicator that the underlying code could benefit from further breakdown or reorganization, which makes the subcomponents easier to test.

Reply:
Yeah, I was torn about testing granularity.  I kinda like having a test like this to demonstrate the full walkthrough of the lookup and having the various edge cases (advantage of the unit tests...they're super fast!).  But I can certainly break apart the methods some more like for the date bit that gets earliest and latest